src/main/java/de/bonndan/nivio/api/ApiController.java

```diff
@@ -90,12 +91,13 @@ public ProcessLog indexLandscape(
         env.setIdentifier(identifier);
         env.setIsPartial(true);
 
-        SourceReference sourceReference = new SourceReference(SourceFormat.from(format));
-        sourceReference.setUrl(null);
+        SourceReference sourceReference = new SourceReference(null, format);
```

Comment:
Right, but there is already a String constructor:

```java
public SourceReference(String url) {
        this.url = url;
    }
```
The url is used to keep track of the origin of the source, i.e. from where it  has been loaded (PULL). The api controller implements PUSH, so we don't know the origin here (?).

Refactoring to SourceReference(format, url) is maybe an option. As always, I'd say this can be done in the development branch.